src/compat/reversediff.jl

```diff
@@ -0,0 +1,78 @@
+const RTR = ReverseDiff.TrackedReal
+const RTV = ReverseDiff.TrackedVector
+const RTM = ReverseDiff.TrackedMatrix
+using ReverseDiff: record_mul
+using ReverseDiff: SpecialInstruction
+
+_eps(::Type{<:RTR{T}}) where {T} = _eps(T)
+
+Base.:*(A::RTM{<:Any, D}, B::AbstractTriangular) where {D} = record_mul(A, B, D)
+Base.:*(A::AbstractTriangular, B::RTV{<:Any, D}) where {D} = record_mul(A, B, D)
+Base.:*(A::AbstractTriangular, B::RTM{<:Any, D}) where {D} = record_mul(A, B, D)
+function Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::RTM{<:Any, D}) where {T, D}
+    return record_mul(A, B, D)
+end
+function Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::RTV{<:Any, D}) where {T, D}
+    return record_mul(A, B, D)
+end
+
+function replace_diag(::typeof(log), X::RTM{<:Any, D}) where {D}
```

Comment:
May I ask why these impls of `replace_diag` are necessary?

Reply:
I couldn't find a simple implementation that maintains a TrackedArray in Tracker and ReverseDiff. So I went with a custom adjoint for this function since it is simple enough to define. This ugly syntax is just a custom adjoint in ReverseDiff.

Reply:
Does indexing with `diagind` not work for TrackedArray and ReverseDiff? I thought it would be problematic only for Zygote due to the mutation.

Reply:
Mutation is not allowed with TrackedArrays too.

Reply:
I see :+1:

Reply:
Ah, sorry; didn't realize `replace_diag` was a function from Bijectors.jl; makes sense :+1: 
Would be awesome with a comment stating why it's necessary:)

Reply:
Sure